### PR TITLE
chore: sign container images on quay and signed releases

### PR DESCRIPTION
**Pull request by 34fathombelow**
Signed releases #9769 
This PR implements container images to be signed using ```sigstore/cosign```  It also consolidates the checksums of the cli-binaries into a single file then signs it.   

#### Three GitHub secrets will need to be created before this PR is merged. The process to do this is listed below.

TLDR: https://docs.sigstore.dev/cosign/git_support

1. [Install Cosign](https://docs.sigstore.dev/cosign/installation) on your workstation. Linux, Homebrew, and container options are available.  Execute the command ```cosign --version``` to verified it has been installed correctly.
2. Create or have a valid GitHub PAT token available.
3. Export your PAT as the environment variable "GITHUB_TOKEN"
4. Execute ```cosign generate-key-pair github://argoproj/argo-workflows``` This will start the process of creating the GitHub Secrets automatically for you, and prompt you to enter a password.  This Password will be stored as the Github secret ```COSIGN_PASSWORD``` I would recommend to use well respected password generator such as KeePassXC or Bitwarden and use a paranoid level of characters of at least 32. 
5. Three GitHub secretes should have been created, please verify.  ```COSIGN_PASSWORD``` ```COSIGN_PRIVATE_KEY``` & ```COSIGN_PUBLIC_KEY```
6. Please comment on this PR with the newly created public key (cosign.pub)

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -388,8 +443,9 @@ jobs:
           body_path: release-notes
           files: |
             dist/argo-*.gz
```
Comment (alexec) [inline .github/workflows/release.yaml:445]: Don’t we want to keep this file?

**Comment by 34fathombelow** [comment]
Updated to only sign multi-image tags

**Review by alexec — APPROVED** [review]
LGTM. Github Actions will need to be monitored when this is merged and settings updated.

**Comment by 34fathombelow** [comment]
Not stale

**Review by terrytangyuan — APPROVED** [review]
Thank you!

**Comment by terrytangyuan** [comment]
We need some additional secrets required by this PR. @alexec @sarabala1979 Could you help add those?

**Pull request merged**